src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestHelpers/TestRunner.cs

```diff
@@ -193,25 +194,47 @@ public void AddFile(string thisFile)
 
                     test.Expected = line.Trim();
 
-                    var key = test.GetUniqueId(fileOveride);
-                    if (_keyToTests.TryGetValue(key, out var existingTest))
+                    foreach (string locale in Locales)
                     {
-                        // Must be in different sources
-                        if (existingTest.SourceFile == test.SourceFile && existingTest.SetupHandlerName == test.SetupHandlerName)
+                        Engine e = _engines.FirstOrDefault(e => e.Config.CultureInfo.Name == locale);
+
+                        if (e == null)
                         {
-                            throw ParseError(i, $"Duplicate test cases in {Path.GetFileName(test.SourceFile)} on line {test.SourceLine} and {existingTest.SourceLine}");
+                            e = new Engine(new PowerFxConfig(new CultureInfo(locale)));
+                            _engines.Add(e);
                         }
 
-                        // Updating an existing test. 
-                        // Inputs are the same, but update the results.
-                        existingTest.MarkOverride(test);
-                    }
-                    else
-                    {
-                        // New test
-                        Tests.Add(test);
+                        TestCase testWithLocale = new TestCase()
+                        {
+                            Input = (locale == "en-US") ? test.Input : e.GetDisplayExpression(test.Input, null as ReadOnlySymbolTable),
```

Comment:
>GetDisplayExpression [](http://example.com/codeflow?start=73&length=20)

What about?

This test case:  
`Value("123,456")`

GetDisplayExpression doesn't convert strings. 

We may have some test holes here, I added this test in a PR: 
[https://github.com/microsoft/Power-Fx/pull/1157](https://github.com/microsoft/Power-Fx/pull/1157)